homework-submission/mobx-week1/src/Componants/TodoList.js

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import { inject, observer } from "mobx-react";
+
+@inject("TodoStore")
+@observer
+class TodoList extends React.Component {
+  state = {
+    clicked: false
+  };
+
+  onEdit = () => {
+    this.setState({
+      clicked: true
+    });
+  };
+  onCancel = () => {
+    this.setState({
+      clicked: false
+    });
+  };
+  updateTodo = (e, id) => {
+    e.preventDefault();
+    const description = this.newDescription.value;
+    const deadline = this.newDeadline.value;
+
+    this.props.TodoStore.updateTodo(id, description, deadline);
+    this.setState({
+      clicked: false
+    });
+  };
+
+  editAction = todo =>
+    this.state.clicked ? (
+      <form className="form-inline" onSubmit={e => this.updateTodo(e, todo.id)}>
+        <input
+          type="text"
+          name="description"
+          className="form-control mb-2 mr-sm-2"
+          placeholder="description"
+          ref={input => (this.newDescription = input)}
+        />
+        <input
+          type="text"
+          name="deadline"
+          className="form-control mb-2 mr-sm-2"
+          placeholder="deadline"
+          ref={input => (this.newDeadline = input)}
+        />
+        <button type="submit" className="btn btn-success">
+          Update
+        </button>
+        <button className="btn btn-warning" onClick={this.onCancel}>
+          Cancel
+        </button>
+      </form>
+    ) : (
+      <span className="todo-text" style={checkboxStyle(todo.done)}>
```

Comment:
Styling is one of the most important things, But what **If** you wanted to change the styling from your stylesheet file in the future.
This won't change, right? 🙃💔

there are many ways of styling improvements in the react world, Have you a look:
https://codeburst.io/4-four-ways-to-style-react-components-ac6f323da822

I saw you are using this function for a multiple times which is great but I would prefer to leave the css to it's own part, up to you, what do you say?